src/horizontal-pod-autoscaler.ts

```diff
@@ -0,0 +1,865 @@
+   *
+   */
+  public static containerStorage(options: MetricContainerResourceOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.CONTAINER_RESOURCE,
+      containerResource: {
+        name: ResourceType.STORAGE,
+        container: options.container.name,
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Metric that tracks the local ephemeral storage of a container. This metric
+   * will be tracked across all pods of the current scale target. Only a single
+   * container may be tracked in each pod. In other words, a pod that hosts two
+   * of the same container, will be ignored.
+   *
+   */
+  public static containerEphemeralStorage(options: MetricContainerResourceOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.CONTAINER_RESOURCE,
+      containerResource: {
+        name: ResourceType.EPHEMERAL_STORAGE,
+        container: options.container.name,
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * A global metric that is not associated with any Kubernetes object.
+   * Allows for autoscaling based on information coming from components running outside of
+   * the cluster.
+   *
+   * Use case:
+   * * Scale up when the length of an SQS queue is greater than 1000 messages.
+   * * Scale down when an outside load balancer's queries are less than 10 per second.
+   */
+  public static external(options: MetricOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.EXTERNAL,
+      external: {
+        metric: {
+          name: options.name,
+          selector: options?.labelSelector?._toKube(),
+        },
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+  * Metric that describes a kubernetes object
+  *
+  * Use case:
+  * * Scale on a Kubernetes Ingress's hits-per-second metric.
+  */
+  public static object(options: MetricObjectOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.OBJECT,
+      object: {
+        describedObject: {
+          apiVersion: options.object.apiVersion,
+          kind: options.object.kind,
+          name: options.object.name,
+        },
+        metric: {
+          name: options.name,
+          selector: options?.labelSelector?._toKube(),
+        },
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * A pod metric that will be averaged across all pods of the current scale target.
+   *
+   * Use case:
+   * * Average CPU utilization across all pods
+   * * Transactions processed per second across all pods
+   */
+  public static pods(options: MetricOptions): Metric {
+    return new Metric({
+      type: MetricSourceType.PODS,
+      pods: {
+        metric: {
+          name: options.name,
+          selector: options?.labelSelector?._toKube(),
+        },
+        target: options.target._toKube(),
+      },
+    });
+  }
+
+  /**
+   * Tracks the available memory of a target.
```

Comment:
```suggestion
   * Tracks the available CPU of the pods in a target.
```

I would also add this note from kubernetes docs:

> Note: Since the resource usages of all the containers are summed up the total pod utilization may not accurately represent the individual container resource usage. This could lead to situations where a single container might be running with high usage and the HPA will not scale out because the overall pod usage is still within acceptable limits.

And the same of the other functions.